### PR TITLE
DRY : Invitation model

### PR DESCRIPTION
**Pull request by ioaNikas**
Added some comments.
Redefined Invitation model to use it in server and client side (contributes to #1043).

### CODE REVIEW

File: libs/organization/src/lib/components/member-pending/member-pending.component.ts [hunk 1]
```diff
@@ -9,8 +9,8 @@ import { Invitation } from '@blockframes/notification';
 })
 
 export class MemberPendingComponent {
-  @Output() accepted = new EventEmitter<Invitation>();
-  @Output() declined = new EventEmitter<Invitation>();
-  @Input() invitations: Invitation[];
+  @Output() accepted = new EventEmitter<InvitationFromUserToOrganization>();
```
Comment (lsenta) [inline libs/organization/src/lib/components/member-pending/member-pending.component.ts]: Are you sure about moving from the generic Invitation type to one specific kind of Invitation? There's no way this code gets another type of invitation?

File: libs/notification/src/lib/invitation/+state/invitation.model.ts [hunk 2]
```diff
@@ -1,61 +1,58 @@
...
+    app: 'main',
+    type: InvitationType.fromOrganizationToUser,
     status: InvitationStatus.pending,
+    date: firestore.Timestamp.now(),
+    ...params
+  };
+}
+
+/** Factory function that create an Invitation of type toWorkOnDocument. */
+export function createInvitationToDocument(params: InvitationToWorkOnDocumentOptions): InvitationToWorkOnDocument {
```
Comment (lsenta) [inline libs/notification/src/lib/invitation/+state/invitation.model.ts]: Would it make sense to have these create functions in the `firestore.ts` file and reuse them in the backend? (IF we create invites there)

**Comment by lsenta** [comment]
Note "issue#1043" won't link the PR, keep spaces: "issue #1043". I updated the PR description to make sure we reference this.

File: libs/organization/src/lib/components/member-pending/member-pending.component.ts (hunk 1 above)
Comment (ioaNikas) [reply libs/organization/src/lib/components/member-pending/member-pending.component.ts]: You are right, this object is going right to a function that takes a basic Invitation. Changing it.

File: libs/notification/src/lib/invitation/+state/invitation.model.ts (hunk 2 above)
Comment (ioaNikas) [reply libs/notification/src/lib/invitation/+state/invitation.model.ts]: Seems like all of our Invitations are generated client-side. Firebase function are only there to react to status change and do operations (like deleting invitation, writing permissions when we accept invitation to a document etc.)

**Pull request merged**